Translations Coverage page

orangejenny opened the pull request:
@mkangia As discussed in the Multimedia Mayhem skype chat while we were debugging https://dimagi-dev.atlassian.net/browse/ICDS-291, I put this together to help verify when references are missing, causing multimedia to fall back to the default language.

<img width="1528" alt="screen shot 2019-02-11 at 2 51 35 pm" src="https://user-images.githubusercontent.com/1486591/52589494-f1a33100-2e0c-11e9-92ef-72c5240b5c98.png">

This is behind the "Bulk multimedia path management" flag for ICDS.
code buddy @dannyroberts

orangejenny committed:
Added page skeleton

orangejenny committed:
Added HTML content

orangejenny committed:
Added skeletal ajax request and UI to handle it

orangejenny committed:
Added real output and switch from ajax to full page reload.

This page isn't especially interactive, so use a normal form submission instead of ajax.

orangejenny committed:
Added default value for media types

orangejenny committed:
Renamed feature flag

orangejenny committed:
Fixed misspelling

orangejenny committed:
Removed unused `for` attribute

mkangia commented on corehq/apps/hqmedia/models.py:
```diff
@@ -972,7 +976,7 @@ def get_reference_totals(self):
         """
         totals = []
-        for mm in [CommCareImage, CommCareAudio, CommCareVideo]:
+        for mm in [CommCareMultimedia.get_doc_class(t) for t in CommCareMultimedia.get_doc_types()]:
```
whats the reason for doing this conversion now?

`CommCareMultimedia` is present in `get_doc_class` but not in `get_doc_types` and that's a little confusing here without the context of why it is skipped.

orangejenny replied:
I just did this to dry up the code, so we don't have multiple lists of all media types (excluding CommCareMultimedia because I don't think that ever gets instantiated, it's just a superclass for the others).

mkangia commented on corehq/apps/hqmedia/templates/hqmedia/translations_coverage.html:
```diff
@@ -0,0 +1,66 @@
+
+{% block js %}{{ block.super }}
+    <script src="{% static 'hqwebapp/js/widgets.js' %}"></script>
+    <script src="{% static 'hqwebapp/js/widgets_v4.js' %}"></script>
```
just confirming that both are needed?

orangejenny replied:
yes

mkangia commented on corehq/apps/hqmedia/views.py:
```diff
@@ -289,6 +293,69 @@ def update_multimedia_paths(request, domain, app_id):
+            "media_types": {t: CommCareMultimedia.get_doc_class(t).get_nice_name()
+                            for t in CommCareMultimedia.get_doc_types()},
+            "selected_langs": self.request.POST.getlist('langs', []),
+            "selected_media_types": self.request.POST.getlist('media_types', ['CommCareAudio', 'CommCareVideo']),
```
`CommCareImage` not included here? Can also use `CommCareMultimedia.get_doc_types`?

orangejenny replied:
My understanding is images typically don't get "translated", just audio and video.

mkangia replied:
haha, that's right :)

mkangia commented on corehq/apps/hqmedia/views.py:
```diff
@@ -289,6 +293,69 @@ def update_multimedia_paths(request, domain, app_id):
+            error = True
+            messages.error(request, "Please select a language.")
+
+        media_types = request.POST.getlist('media_types')
```
Should not one be allowed to check for all media types if none are provided?

orangejenny replied:
Hmm...I see where you're coming from, but it isn't a pattern we use elsewhere, and there are at most 3 types to pick, so I'm going to leave this as-is.

mkangia reviewed (approved):
Will it be useful to extend this to allow the user to check for a specific version of the application to find the missing coverage in a released build?

nothing blocking.

orangejenny commented:
> Will it be useful to extend this to allow the user to check for a specific version of the application to find the missing coverage in a released build?

Good thought, I'll look into that.

orangejenny committed:
Added search by version to releases search

orangejenny committed:
Updated try to use else

orangejenny committed:
Moved re-sort and re-limit to only happen if build was added to list

orangejenny committed:
Renamed build_comment to query

orangejenny committed:
Updated version dropdown on source files page to use ajax

orangejenny committed:
Renamed js variable

orangejenny committed:
Removed unused built_versions context param

orangejenny committed:
Merge branch 'jls/ajaxify-version-dropdown' into jls/trans-coverage

orangejenny committed:
Added version dropdown

orangejenny committed:
Merge branch 'jls/trans-coverage' of github.com:dimagi/commcare-hq into jls/trans-coverage

orangejenny commented:
@mkangia I added the version dropdown in 8c0f2ae084e6d265f00728b8170a0339e5654e7d

Added don't merge label because this now depends on https://github.com/dimagi/commcare-hq/pull/23242 and https://github.com/dimagi/commcare-hq/pull/23289 being merged.

<img width="1454" alt="screen shot 2019-02-19 at 12 05 55 pm" src="https://user-images.githubusercontent.com/1486591/53033427-02702a00-343f-11e9-9fc5-ffc86feed6a0.png">

mkangia reviewed (approved):
just something around naming.

orangejenny committed:
Added compress tag

orangejenny committed:
Added comment to currently selected build

mkangia commented on corehq/apps/hqmedia/views.py:
```diff
@@ -318,7 +318,9 @@ def page_context(self):
-            selected_build_text = build.version
+            selected_build_text = str(build.version)
+            if build.build_comment:
+                selected_build_text += ": " + build.build_comment
```
👍

mkangia commented on corehq/apps/hqmedia/templates/hqmedia/translations_coverage.html:
```diff
@@ -1,11 +1,14 @@
+        <script src="{% static 'hqwebapp/js/widgets.js' %}"></script>
+        <script src="{% static 'hqwebapp/js/widgets_v4.js' %}"></script>
+        <script src="{% static 'app_manager/js/widgets_v4.js' %}"></script>
+    {% endcompress js %}
```
👍

orangejenny committed:
Merge branch 'master' into jls/trans-coverage
